hydra-node/test/Hydra/Model.hs

```diff
@@ -406,16 +368,40 @@ runModel = RunModel{perform}
       Init party contestationPeriod ->
         party `sendsInput` Input.Init{contestationPeriod}
       Abort party -> do
-        party `sendsInput` Input.Abort
-      Wait timeout ->
-        lift $ threadDelay timeout
+        performAbort party
+      Wait delay ->
+        lift $ threadDelay delay
       ObserveConfirmedTx tx -> do
         nodes <- Map.toList <$> gets nodes
-        forM_ nodes $ \(party, node) -> do
-          party `sendsInput` Input.GetUTxO
-          waitForUTxOToSpend mempty (to tx) (value tx) party node 1000 >>= \case
+        forM_ nodes $ \(_, node) -> do
+          lift (waitForUTxOToSpend mempty (to tx) (value tx) node 1000000) >>= \case
             Left u -> error $ "Did not observe transaction " <> show tx <> " applied: " <> show u
             Right _ -> pure ()
+      StopTheWorld ->
+        stopTheWorld
+
+performAbort :: MonadDelay m => Party -> StateT (Nodes m) m ()
+performAbort party = do
+  Nodes{nodes} <- get
+  lift $ waitForReadyToCommit party nodes 100
+  party `sendsInput` Input.Abort
+
+waitForReadyToCommit :: MonadDelay m => Party -> Map Party (TestHydraNode tx m) -> Int -> m ()
+waitForReadyToCommit _party _nodes 0 = pure ()
+waitForReadyToCommit party nodes n = do
+  outs <- serverOutputs (nodes ! party)
```

Comment:
I guess there are other usages of `serverOutputs` having the same problem.

Reply:
I am not too sure actually, at one point I had this issue whereby I lost outputs because there was another `waitMatch`. Perhaps it's not an issue anymore

Reply:
I tried and it hangs because when we start waiting for match, the output we are looking for might have been already sent, this is why I reverted to the admittedly lame `serverOutputs`